File: PracticaParcialTS/PDF2/Ejercicio3.ts
//Catálogo de artículos de venta
abstract class Articulo {
    abstract mostrar(): string;
}

class ArticuloTexto extends Articulo {
    constructor(private texto: string) {
        super();
    }

    mostrar(): string {
        return `Texto: ${this.texto}`;
    }
}

class ArticuloImagen extends Articulo {
    constructor(private imagen: string) {
        super();
    }

    mostrar(): string {
        return `Imagen: ${this.imagen}`;
    }
}

class ArticuloVideo extends Articulo {
    constructor(private video: string) {
        super();
    }

    mostrar(): string {
        return `Video: ${this.video}`;
    }
}

// Ejemplo de uso
const articulos: Articulo[] = [
    new ArticuloTexto("Descripción de un producto"),
    new ArticuloImagen("imagen_producto.jpg"),
    new ArticuloVideo("video_producto.mp4")
];

articulos.forEach(articulo => {
    console.log(articulo.mostrar());
});

File: PracticaDeEjerciciosParcial/Ejercicio3.py
#Catálogo de artículos de venta
class Articulo(ABC):
    @abstractmethod
    def mostrar(self):
        pass

class ArticuloTexto(Articulo):
    def __init__(self, texto):
        self.texto = texto

    def mostrar(self):
        return f"Texto: {self.texto}"

class ArticuloImagen(Articulo):
    def __init__(self, imagen):
        self.imagen = imagen

    def mostrar(self):
        return f"Imagen: {self.imagen}"

class ArticuloVideo(Articulo):
    def __init__(self, video):
        self.video = video

    def mostrar(self):
        return f"Video: {self.video}"

# Ejemplo de uso
articulos = [
    ArticuloTexto("Descripción de un producto"),
    ArticuloImagen("imagen_producto.jpg"),
    ArticuloVideo("video_producto.mp4")
]

for articulo in articulos:
    print(articulo.mostrar())
